### ISSUE TITLE
[1.5.0] Migration from version 1.4.0

### ISSUE THREAD

**Issue by afuno** [github/actionview-component#122]
Could you please tell about migration from version 1.4.0?

I updated the gem.

I changed this line:

```ruby
require 'action_view/component/base'
```

To this:

```ruby
require 'action_view/component'
```

After that, I can't start the project:

```
% rails s
=> Booting Puma
=> Rails 6.0.1 application starting in development
=> Run `rails server --help` for more startup options
Exiting
Traceback (most recent call last):
	77: from bin/rails:5:in `<main>'
	76: from bin/rails:5:in `load'
	75: from /Users/afuno/Projects/my.app/bin/spring:16:in `<top (required)>'
	74: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	73: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	72: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/spring-2.1.0/lib/spring/binstub.rb:11:in `<top (required)>'
	71: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/spring-2.1.0/lib/spring/binstub.rb:11:in `load'
	70: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/spring-2.1.0/bin/spring:49:in `<top (required)>'
	69: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/spring-2.1.0/lib/spring/client.rb:30:in `run'
	68: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/spring-2.1.0/lib/spring/client/command.rb:7:in `call'
	67: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/spring-2.1.0/lib/spring/client/rails.rb:28:in `call'
	66: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/spring-2.1.0/lib/spring/client/rails.rb:28:in `load'
	65: from /Users/afuno/Projects/my.app/bin/rails:11:in `<top (required)>'
	64: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/dependencies.rb:325:in `require'
	63: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/dependencies.rb:291:in `load_dependency'
	62: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/dependencies.rb:325:in `block in require'
	61: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	60: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
	59: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	58: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
	57: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
	56: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/commands.rb:18:in `<main>'
	55: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/command.rb:46:in `invoke'
	54: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/command/base.rb:65:in `perform'
	53: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	52: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	51: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	50: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/commands/server/server_command.rb:138:in `perform'
	49: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/commands/server/server_command.rb:138:in `tap'
	48: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/commands/server/server_command.rb:147:in `block in perform'
	47: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/commands/server/server_command.rb:37:in `start'
	46: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/commands/server/server_command.rb:77:in `log_to_stdout'
	45: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/server.rb:354:in `wrapped_app'
	44: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/server.rb:219:in `app'
	43: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/server.rb:319:in `build_app_and_options_from_config'
	42: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/builder.rb:40:in `parse_file'
	41: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/builder.rb:49:in `new_from_string'
	40: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/builder.rb:49:in `eval'
	39: from config.ru:in `<main>'
	38: from config.ru:in `new'
	37: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/builder.rb:55:in `initialize'
	36: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/rack-2.0.7/lib/rack/builder.rb:55:in `instance_eval'
	35: from config.ru:5:in `block in <main>'
	34: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:48:in `require_relative'
	33: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/dependencies.rb:325:in `require'
	32: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/dependencies.rb:291:in `load_dependency'
	31: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/dependencies.rb:325:in `block in require'
	30: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/zeitwerk-2.2.1/lib/zeitwerk/kernel.rb:23:in `require'
	29: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	28: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
	27: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	26: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
	25: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
	24: from /Users/afuno/Projects/my.app/config/environment.rb:7:in `<main>'
	23: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/application.rb:363:in `initialize!'
	22: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/initializable.rb:60:in `run_initializers'
	21: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/tsort.rb:205:in `tsort_each'
	20: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/tsort.rb:226:in `tsort_each'
	19: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/tsort.rb:347:in `each_strongly_connected_component'
	18: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/tsort.rb:347:in `call'
	17: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/tsort.rb:347:in `each'
	16: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/tsort.rb:349:in `block in each_strongly_connected_component'
	15: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/tsort.rb:431:in `each_strongly_connected_component_from'
	14: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	13: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/2.6.0/tsort.rb:228:in `block in tsort_each'
	12: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/initializable.rb:61:in `block in run_initializers'
	11: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/initializable.rb:32:in `run'
	10: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/initializable.rb:32:in `instance_exec'
	 9: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/railties-6.0.1/lib/rails/application/finisher.rb:129:in `block in <module:Finisher>'
	 8: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/lazy_load_hooks.rb:51:in `run_load_hooks'
	 7: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/lazy_load_hooks.rb:51:in `each'
	 6: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/lazy_load_hooks.rb:52:in `block in run_load_hooks'
	 5: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/lazy_load_hooks.rb:67:in `execute_hook'
	 4: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/lazy_load_hooks.rb:62:in `with_execution_control'
	 3: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.1/lib/active_support/lazy_load_hooks.rb:69:in `block in execute_hook'
	 2: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/actionview-component-1.5.0/lib/action_view/component/railtie.rb:52:in `block in <class:Railtie>'
	 1: from /Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/change_observer.rb:10:in `<<'
/Users/afuno/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/change_observer.rb:10:in `<<': can't modify frozen #<Class:#<Array:0x00007ff535a70be8>> (FrozenError)
```

**Comment by joelhawksley** [comment]
@afuno thanks for the report, you're on top of things!

Might you be able to provide a PR with a failing test?

**Comment by afuno** [comment]
@joelhawksley I don't know how to do PR. I have a project in which I took these two steps. And after that I can't start it. In version 1.4.0 everything is fine.

**Comment by joelhawksley** [comment]
@afuno it might be easier to provide a demo Rails application in a new repo that shows this issue.

I'm going to guess that it's a compatibility problem with Bootsnap.

**Comment by afuno** [comment]
@joelhawksley You have access to this repository :)

**Comment by joelhawksley** [comment]
@afuno I'm unable to reproduce the issue in the repository. The server boots with `rails s` and the page loads.

**Comment by afuno** [comment]
@joelhawksley Could this be somehow related to a point in the path to the project directory? I don't know, just the first thought :)

**Comment by afuno** [comment]
Deleting all components does not help.
This problem is also present in my other applications, in which everything was fine before.

**Comment by juanmanuelramallo** [comment]
Just found the issue will try to open a PR shortly 
It's in our railties. 
```ActiveSupport::Dependencies.autoload_paths``` That's the  frozen array it seems

**Comment by franciscoj** [comment]
I have a small patch for it, I can share it in case you think it can be of help.

**Comment by juanmanuelramallo** [comment]
Please @franciscoj

**Comment by franciscoj** [comment]
```
diff --git a/lib/action_view/component/railtie.rb b/lib/action_view/component/railtie.rb
index 422614a..cb57c7d 100644
--- a/lib/action_view/component/railtie.rb
+++ b/lib/action_view/component/railtie.rb
@@ -39,6 +39,14 @@ module ActionView
         end
       end

+      initializer "action_view_component.set_preview_autoload_paths" do |app|
+        options = app.config.action_view_component
+
+        if options.preview_path
+          ActiveSupport::Dependencies.autoload_paths << options.preview_path
+        end
+      end
+
       config.after_initialize do |app|
         options = app.config.action_view_component

@@ -47,10 +55,6 @@ module ActionView
             get "/rails/components" => "rails/components#index", :internal => true
             get "/rails/components/*path" => "rails/components#previews", :internal => true
           end
-
-          if options.preview_path
-            ActiveSupport::Dependencies.autoload_paths << options.preview_path
-          end
         end
       end
     end
```

**Comment by juanmanuelramallo** [comment]
Amazing thank you!

**Comment by franciscoj** [comment]
I managed to reproduce the issue from @afuno in an app with rails 6.0.1 by using Zeitwerk.

The issue disappears if you disable Zeitwerk with:

```
    config.autoloader = :classic
```

on `config/application.rb`

**Comment by juanmanuelramallo** [comment]
Yep, it was working fine for me in a Rails 5.2 app. 

But as soon as I switched to Rails 6, I saw the issue. 

This should solve it https://github.com/github/actionview-component/pull/124
Thanks for the patch @franciscoj

**Comment by franciscoj** [comment]
You're welcome!

**Comment by afuno** [comment]
@joelhawksley It seems that the problem is resolved in version 1.5.1. Thanks.

**Comment by joelhawksley** [comment]
Marking this as resolved. Thanks for the fix @juanmanuelramallo!
